### PR TITLE
bartycrouch: build with swift on Catalina/Linux

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

See Homebrew/brew#11964, #84559.

### CODE REVIEW

**Comment by carlocab** [comment]
Did this just fail to write somewhere and not know how to handle it?

**Comment by Bo98** [comment]
The lack of a Darwin SDK overlay is potentially a problem. I stopped shipping all of them but I think we might want to keep Darwin (that's actually going to be the only one left with Swift 5.5).

But I expected it to work since there is a Darwin Swift module in the SDK since Catalina. Maybe there's a mixup of SDKs here - Homebrew will be trying to use the Catalina SDK in the shims but SwiftPM picks up the Big Sur one here (adjustable by `ENV["SDKROOT"] = ENV["HOMEBREW_SDKROOT"]` probably).

File: Formula/bartycrouch.rb [hunk 1]
```diff
@@ -12,10 +12,27 @@ class Bartycrouch < Formula
...
 
-  depends_on xcode: ["12.5", :build]
-  depends_on :macos
+  if MacOS.version >= :big_sur
+    depends_on xcode: ["12.5", :build]
+  else
+    depends_on "swift" => :build
+  end
+
+  uses_from_macos "swift"
```
Comment (carlocab) [inline Formula/bartycrouch.rb:21]:
```suggestion
    depends_on "swift"
  end
```

**Comment by Bo98** [comment]
Ah it so happens we managed to pick another formula using SwiftSyntax?

**Pull request closed without merging**